069-shared-volume-mount.md

```diff
@@ -0,0 +1,108 @@
+It only allows mounting of configmaps and secrets. So if there are any files written to disk (by sink file connectors) then will  be lost
+if the pod restarts. This limits task resiliency in distributed environments, which is a critical aspect of maintaining high availability and fault tolerance in such systems.
+
+## Motivation
+
+The primary motivation for this proposed enhancement to the Strimzi Kafka Operator is to ensure uninterrupted access to data across all kafka connect pods in a distributed cluster. In the current setup, if a pod goes down, the data stored in its mount path becomes inaccessible, potentially disrupting operations. By enabling shared volume mounts across all pods, we can ensure that even if one pod is down, the data remains accessible from other pods. This feature is crucial for maintaining high availability and fault tolerance, thereby significantly improving the resilience and efficiency of tasks of connectors. Hence, this proposed change is not just an improvement, but a necessity for robust and seamless data management in distributed kafka connect clusters.
+ 
+Support for resilliency for sink file connnectors: Mounting volumes will allow a task that gets reallocted to a different pod to continue processing the file. Also allows for exactly once  sink file connectors to store their offsets to file without them being lost in case  of pod restart.
+Support for Pod Running Logs: Ability to dump kafka connect logs to storage volume instead of just stdout.
+GC Logs Management: Implement a JVM option to output GC logs to a file instead of stdout when GC logging is enabled. This will help in identifying JVM hangs and keep stdout uncluttered.
+Additional diagnostics: Explore the possibility of mounting additional volumes into the Kafka Connect pods beyond ConfigMaps or Secrets. This could facilitate writing event data to parquet files for later analysis.
+
+## Proposal
+
+This change introduces enhancements to the Strimzi Kafka Operator for managing shared volume mounts, improving task resiliency across nodes. Key changes include:
+
+1) **mountPath:** A string property to specify the container directory for mounting the storage volume. While the default is /var/lib/kafka/data, this allows for custom path specification. By default, it will try to mount read write many storage if accessMode is not specified.
+
+2) **accessMode:** An optional string property defining the storage access mode, determining permissions like read-only or read-write.
+
+These changes, tested successfully with Read-Write-Many on Ceph-block storage and NFS, provide greater control over volume configuration and usage, improving operator performance in distributed environments. 
+
+**Sample Test**
+
+ - This test case is to test the strimzi operator customization that allows mounting of shared volume across kafka-connect pods 
```

Comment:
You should also explain how will this be cover in the various test environments -> how will you provide the shared storage for example?